server/src/main/java/org/apache/druid/server/coordinator/rules/LoadRule.java

```diff
@@ -144,14 +144,15 @@ private void assign(
   private static List<ServerHolder> getFilteredHolders(
       final String tier,
       final DruidCluster druidCluster,
-      final Predicate<ServerHolder> predicate
+      Predicate<ServerHolder> predicate
   )
   {
     final NavigableSet<ServerHolder> queue = druidCluster.getHistoricalsByTier(tier);
     if (queue == null) {
       log.makeAlert("Tier[%s] has no servers! Check your cluster configuration!", tier).emit();
       return Collections.emptyList();
     }
+    predicate = predicate.and(s -> !s.isMaintenance());
```

Comment:
Makes sense to put `isMaintenance` check first because it's definitely cheap.

Reply:
:white_check_mark: